packages/@aws-cdk/aws-appsync/lib/graphqlapi-base.ts

```diff
@@ -0,0 +1,193 @@
+import { ITable } from '@aws-cdk/aws-dynamodb';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { CfnResource, IResource, Resource } from '@aws-cdk/core';
+import { DynamoDbDataSource, HttpDataSource, LambdaDataSource, NoneDataSource } from './data-source';
+
+/**
+ * Optional configuration for data sources
+ */
+export interface DataSourceOptions {
+  /**
+   * The name of the data source, overrides the id given by cdk
+   *
+   * @default - the id provisioned
+   */
+  readonly name?: string;
+
+  /**
+   * The description of the data source
+   *
+   * @default - No description
+   */
+  readonly description?: string;
+}
+
+/**
+ * Interface for GraphQL
+ */
+export interface IGraphqlApi extends IResource {
+
+  /**
+   * an unique AWS AppSync GraphQL API identifier
+   * i.e. 'lxz775lwdrgcndgz3nurvac7oa'
+   *
+   * @attribute
+   */
+  readonly apiId: string;
+
+  /**
+   * the ARN of the API
+   *
+   * @attribute
+   */
+  readonly arn: string;
+
+  /**
+   * add a new dummy data source to this API. Useful for pipeline resolvers
+   * and for backend changes that don't require a data source.
+   *
+   * @param id The data source's id
+   * @param options The optional configuration for this data source
+   * @default name - the id
```

Comment:
I don't believe `@default` is needed for these methods - but if you did include it, describe the default behavior from not providing any `options`

Reply:
sweet okay i can remove them